issue-sync-helper.sh enrich: unbound _state_json when ENRICH_PREFETCH_FILE not set (regression from #20133)

## Description

`issue-sync-helper.sh enrich <task-id>` fails immediately with `line 1231: _state_json: unbound variable` when `ENRICH_PREFETCH_FILE` is not set in the environment.

The batch `cmd_enrich` flow (which runs `_enrich_prefetch_issues_map` first to populate the env var) is unaffected. The **single-task direct-invocation path** — the common case for interactive users who updated a brief and want to re-sync the GitHub issue body — is broken.

## Expected Behavior

`issue-sync-helper.sh enrich <task-id>` should re-fetch the issue, apply the current brief's title and body, and update. It should do so whether or not a prefetch file was populated by a prior batch call, since the fallback `gh issue view` at line 1232 exists precisely for this case.

## Reproducer

**Symptom command**:

```bash
issue-sync-helper.sh enrich <task-id> --force
```

**Actual output**:

```
[INFO] Enriching 1 issue(s) in <repo>
/home/rob/.aidevops/agents/scripts/issue-sync-helper.sh: line 1231: _state_json: unbound variable
```

**Expected output**: successful enrich with the issue body updated to match the post-edit brief.

**Causal code** (`.agents/scripts/issue-sync-helper.sh`):

```bash
# Line 27
set -euo pipefail

# Line 1219 — _state_json declared WITHOUT "" while siblings get explicit init
local _state_json current_title="" current_body="" current_labels_csv=""

# Lines 1224-1227 — conditional assignment, fires only if prefetch file exists
if [[ -n "${ENRICH_PREFETCH_FILE:-}" && -f "$ENRICH_PREFETCH_FILE" && -n "$num" ]]; then
    _state_json=$(jq -c --argjson n "$num" '.[] | select(.number == $n)' \
        "$ENRICH_PREFETCH_FILE" 2>/dev/null || echo "")
fi

# Line 1231 — unbound variable error under set -u when prefetch path skipped
if [[ -z "$_state_json" ]]; then
    _state_json=$(gh issue view "$num" --repo "$repo" --json title,body,labels,state,assignees 2>/dev/null || echo "")
fi
```

**Regression commit**:

```
git blame -L 1219,1226 .agents/scripts/issue-sync-helper.sh
# e00eb5464d (2026-04-18) line 1219 — original declaration (unaffected)
# 5388d0824b (2026-04-20) lines 1220-1226 — added conditional prefetch
#   commit: fix(enrich): add rate-limit probe + batch prefetch to eliminate
#           84% GUARD_UNCERTAIN skip rate (#20133)
```

The prefetch commit introduced the conditional assignment but left `_state_json` without an empty default on line 1219. When `ENRICH_PREFETCH_FILE` is unset (the single-task direct-invocation case), the variable is unbound at line 1231.

## Proposed Fix

**Option A (preferred, one character, matches sibling vars on same line)**:

```diff
-    local _state_json current_title="" current_body="" current_labels_csv=""
+    local _state_json="" current_title="" current_body="" current_labels_csv=""
```

**Option B (parameter expansion default at the check site)**:

```diff
-    if [[ -z "$_state_json" ]]; then
+    if [[ -z "${_state_json:-}" ]]; then
```

Option A is preferred — it matches the initialisation pattern of the three adjacent vars and keeps the guard readable at line 1231.

## Call-site Sweep

Other multi-var `local` declarations in the same file were checked (`rg -n '^\s*local\s+\w+\s+\w+' .agents/scripts/issue-sync-helper.sh`):

- L225 `local _t _r` — `_t` set by for-loop, `_r` assigned before read
- L648 `local url gh_exit combined` — all three assigned in the following block
- L903 `local _lbl_chk _found_chk` — `_lbl_chk` set by for-loop, `_found_chk` assigned before read
- L1066 `local _rl_json _rl_remaining _rl_reset _rc=0` — all set before read
- L1322, L1581 — all set before read

L1219 is the only occurrence where a declared variable is read under `set -u` via a conditional assignment path that can be skipped. No further instances to fix.

## Steps to Reproduce

1. In an interactive session with a TODO.md + brief-synced issue.
2. Edit a brief (e.g., title change) and commit.
3. Run `issue-sync-helper.sh enrich <task-id> --force` without a preceding batch `cmd_enrich` in the same shell (i.e., `ENRICH_PREFETCH_FILE` is empty/unset).
4. Observe the unbound-variable error; issue body is not updated.

## Workarounds Applied

- **Manual body patch via `gh issue edit --body-file`**: fetched current body with `gh issue view --json body`, applied the brief edits via a Python text-replacement pass, pushed back. Adds ~5 min to every single-task re-sync. No user-visible command exists today for "re-sync my brief to the issue" that doesn't hit this path.

## Systemic Observation

`cmd_enrich`/`_enrich_task` have now shipped two unbound-variable regressions (this one + #1681, t1122 "IFS unbound variable error in cmd_enrich" fixed 2026-02-19). The immediate fix is one character. The systemic ask is whether a CI test exercising the single-task path without `ENRICH_PREFETCH_FILE`, or a shellcheck `SC2154` ratchet on this file, is warranted. Noting only — not filing a follow-up unless maintainer signals interest.

## Environment

- **aidevops version**: 3.8.96
- **Latest version**: 3.8.96
- **AI Assistant**: OpenCode (per `log-issue-helper.sh diagnostics`; actual runtime is Claude Code — possibly a separate detection glitch, out of scope for this issue)
- **OS**: Ubuntu 24.04.4 LTS
- **Shell**: bash 5.2.21(1)-release
- **Working repo**: a managed private repo (main)
- **gh CLI**: gh version 2.90.0 (2026-04-16)

## Additional Context

- Prior art: #1681 (t1122) — same class of bug, different variable, same function. Fixed 2026-02-19.
- Observed while re-syncing an issue body after a narrow brief edit in a session where the batch `cmd_enrich` had not run.

<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.8.96 plugin for [OpenCode](https://opencode.ai) v1.14.22 with claude-opus-4-7 spent 1h 59m and 106,369 tokens on this with the user in an interactive session.

## Review: APPROVE — ship Option A

Verified the bug reproduces exactly as described. One-character fix at the declaration site, matches local convention, no conflicts with any open work.

### Validation

| Check | Result |
|-------|--------|
| Reproducible | Yes — `set -u` + skipped conditional assignment + read at `:1231` = guaranteed unbound error on the direct-invocation path |
| Line refs match HEAD | Yes — `:1219` quoted matches HEAD verbatim |
| Pre-existing duplicate | No — #1681 was same class (`IFS` unbound) but different variable, already fixed |
| Superseded by recent work | No — no PRs touch `:1219` since regression commit `5388d08` (#20133, 2026-04-20) |
| Actual bug | Yes — single-task `enrich <task-id>` without `ENRICH_PREFETCH_FILE` is fully broken |

### Root cause

Commit `5388d08` (#20133 "batch prefetch") added a conditional assignment guarded by `[[ -n "${ENRICH_PREFETCH_FILE:-}" ... ]]` but didn't update the declaration at `:1219` to initialize the variable. Under `set -u`, the fallback read at `:1231` triggers the unbound error whenever prefetch is unset — which is every direct-invocation call.

### Fix — Option A (recommended)

```diff
-    local _state_json current_title="" current_body="" current_labels_csv=""
+    local _state_json="" current_title="" current_body="" current_labels_csv=""
```

Why A over B (`${_state_json:-}` at the check site):
- Fixes root cause at the declaration, not the symptom at the read
- Matches the existing pattern of the three adjacent variables on the same line
- One character

### Conflicts / risks

**None identified.**

- No open issue or PR touches this code path
- Pure bug fix in a helper — no impact on approval / sandbox / dedup / privacy / review-bot gates
- Rollback = trivial revert
- Reporter's call-site sweep (`:225`, `:648`, `:903`, `:1066`, `:1322`, `:1581`) is correct — `:1219` is the only skippable-conditional-read path in the function

### Dispatchability

- Issue body is worker-ready per t2417 heuristic (What / Why / How / Acceptance / exact diff present) — no separate brief file needed
- Qualifies for `tier:simple`: 1 file, exact `oldString`/`newString`, <50 lines context, no judgment calls

To dispatch after approval:

1. `sudo aidevops approve issue 20636` (clears `needs-maintainer-review`)
2. `claim-task-id.sh` creates TODO entry + `tNNN` for traceability
3. Apply `tier:simple` + `auto-dispatch`

Verification command:

```bash
bash -c 'unset ENRICH_PREFETCH_FILE; .agents/scripts/issue-sync-helper.sh enrich <any-task-id> --force'
```

Should complete without `_state_json: unbound variable`.

### Systemic observation (non-blocking follow-up)

Two unbound-variable regressions have now shipped in this same function in ~2 months (#1681, now this). Fixes take seconds; the gap is preventive CI. Worth filing as a separate task after this lands:

1. **`shellcheck -o all` with SC2154 ratchet gate** on `issue-sync-helper.sh` — catches unset-read under `set -u` at check time
2. **Integration test** exercising direct-invocation without `ENRICH_PREFETCH_FILE` — catches the class entirely

Not blocking this fix.

---

Textbook bug report — clear reproduction, correct root cause, minimal fix, correct scope sweep. Ready for approval.

<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.8.96 plugin for [OpenCode](https://opencode.ai) v1.14.22 with claude-opus-4-7 spent 7m and 3,525 tokens on this with the user in an interactive session. Overall, 10m since this issue was created.

--- a/.agents/configs/simplification-state.json
+++ b/.agents/configs/simplification-state.json
@@ -2544,9 +2544,9 @@
       "pr": 15916
     },
     ".agents/scripts/shared-constants.sh": {
-      "at": "2026-04-22T23:56:55Z",
-      "hash": "8a03b8f8e70a61bda90e2d4cc41999b14310ecd6",
-      "passes": 20,
+      "at": "2026-04-23T23:17:00Z",
+      "hash": "30b46728d9b1dd3d13e673860c43a4f9b8e68236",
+      "passes": 21,
       "pr": 18847
     },
     ".agents/scripts/stats-functions.sh": {
@@ -7486,9 +7486,9 @@
       "pr": 16415
     },
     "TODO.md": {
-      "at": "2026-04-23T21:03:24Z",
-      "hash": "5fa000f90be603deae873480ef3bbedb178c8a4c",
-      "passes": 92,
+      "at": "2026-04-23T23:19:53Z",
+      "hash": "57a63b74a85455e876fe331cd7dad4931dad56df",
+      "passes": 93,
       "pr": 15490
     },
     "aidevops.sh": {
@@ -7576,10 +7576,10 @@
       "passes": 1
     },
     ".agents/reference/shell-style-guide.md": {
-      "hash": "a611a5a02dbd5e29c9813543ef3546f1414608fe",
-      "at": "2026-04-16T21:58:05Z",
+      "hash": "0a2560bc09c5304973263f831563e4add4c13ab3",
+      "at": "2026-04-23T23:16:35Z",
       "pr": 19321,
-      "passes": 1
+      "passes": 2
     },
     ".agents/scripts/tests/test-consolidation-multi-runner.sh": {
       "hash": "a13a53c49a0bb67c640682bee85ec6eca8c9338a",

--- a/.agents/scripts/issue-sync-helper.sh
+++ b/.agents/scripts/issue-sync-helper.sh
@@ -1216,7 +1216,7 @@ _enrich_process_task() {
 	# sync-on-push 10-minute cap. Forwarding pre-fetched state collapses the
 	# per-task read cost to one call and lets each helper skip writes whose
 	# target value already matches.
-	local _state_json current_title="" current_body="" current_labels_csv=""
+	local _state_json="" current_title="" current_body="" current_labels_csv=""
 	# GH#20129: use batch-prefetched JSON when available to avoid per-task API
 	# calls. ENRICH_PREFETCH_FILE is set by cmd_enrich before the loop via
 	# _enrich_prefetch_issues_map. The prefetch includes all fields needed:

